### PR TITLE
Update to yacman 0.9.3 to prepare for the new YAMLConfigManager

### PR DESCRIPTION
**Pull request by nsheff**
Work toward accommodating yacman v1.

Update the way YAMLConfigManager objects are constructed, read, and written following the upgrade guide in the yacman readme: https://github.com/databio/yacman

### CODE REVIEW

**Comment by simeoncarstens** [comment]
Currently, `looper` uses a (at least, according to the comment) copy & pasted `divvy` module, which in turn uses `YAMLConfigurationManager` (https://github.com/pepkit/looper/blob/1468956dde66abf5b853c80eaeaee2d411bfad64/looper/divvy.py#L31). That would need to be adapted, too.

**Comment by donaldcampbelljr** [comment]
I've finished implementing the FutureYamlConfigManager in this PR but it needs review.

**Review by khoroshevskyi — COMMENTED** [review]
I suggest to discuss this changes with other people

File: requirements/requirements-all.txt [hunk 1]
```diff
@@ -6,8 +6,8 @@ logmuse>=0.2.0
 pandas>=2.0.2
 pephubclient>=0.1.2
 peppy>=0.40.0
-pipestat>=0.8.0
+pipestat>=0.8.2a1
```
[the remark this answers is not in the document]
Comment (donaldcampbelljr) [reply requirements/requirements-all.txt:9]: This is going to dev branch. I figured we would release `pipestat 0.8.2` before the next looper release.

File: looper/divvy.py [hunk 2]
```diff
@@ -47,35 +47,59 @@ class ComputingConfiguration(YAMLConfigManager):
...
+                entries=entries, schema_source=DEFAULT_CONFIG_SCHEMA
+            )
+        else:
+            raise Exception(
+                "Must provide either filepath or entries when creating ComputingConfiguration."
+            )
+
+        self.data.update(temp_ym.data)
+        setattr(self, "schema", temp_ym.schema)
+        self.schema_source = temp_ym.schema_source
```
[the remark this answers is not in the document]
Comment (donaldcampbelljr) [reply looper/divvy.py:82]: I agree that this seems non-ideal. My other approaches were dead ends, unfortunately. Perhaps, I missed something?

File: tests/divvytests/conftest.py [hunk 3]
```diff
@@ -3,23 +3,29 @@
...
+dcc_filepath = select_divvy_config(None)
+DCC = divvy.ComputingConfiguration.from_yaml_file(filepath=dcc_filepath)
+DCC_ATTRIBUTES = DCC.keys()
 
 
 @pytest.fixture
 def empty_dcc():
     """Provide the empty/default ComputingConfiguration object"""
-    return divvy.ComputingConfiguration()
+    return divvy.ComputingConfiguration.from_yaml_file(filepath=dcc_filepath)
```
Comment (nsheff) [inline tests/divvytests/conftest.py:22]: this doesn't look like an empty DCC, is this correct?

**Comment by nsheff** [comment]
looks much cleaner!

File: tests/divvytests/conftest.py (hunk 3 above)
Comment (donaldcampbelljr) [reply tests/divvytests/conftest.py]: Fixed!

**Pull request merged**